src/extension/wps/src/main/java/au/org/emii/gogoduck/worker/GoGoDuck.java

```diff
@@ -291,100 +282,121 @@ public void run() {
-        userLog.log(String.format("Applying subset '%s'", module.getNcksSubsetParameters().toString()));
         logger.info(String.format("Applying subset on directory '%s'", tmpDir));
+        List<String> ncksSubsetParameters;
 
-        File[] directoryListing = tmpDir.toFile().listFiles();
-        logger.info(String.format("Subset for operation is '%s'", module.getNcksSubsetParameters()));
-
-        ExecutorService executorService = Executors.newFixedThreadPool(threadCount);
-        for (final File file : directoryListing) {
-            executorService.submit(new NcksRunnable(file, module, this));
-        }
-
-        executorService.shutdown();
         try {
+            File[] directoryListing = tmpDir.toFile().listFiles();
+            ExecutorService executorService = Executors.newFixedThreadPool(threadCount);
+
+            if (directoryListing != null) {
+                for (File file : directoryListing) {
+                    ncksSubsetParameters = module.getNcksSubsetParameters().getNcksParameters();
+                    logger.info(String.format("Subset for operation is '%s'", ncksSubsetParameters));
+                    userLog.log(String.format("Applying subset '%s'", ncksSubsetParameters));
+                    executorService.submit(new NcksRunnable(file, module, this));
+                }
+            } else {
+                throw new GoGoDuckException("No files available for sub-setting");
+            }
+
+            executorService.shutdown();
             executorService.awaitTermination(MAX_EXECUTION_TIME_DAYS, TimeUnit.DAYS);
-        }
-        catch (InterruptedException e) {
+        } catch (InterruptedException e) {
             throw new GoGoDuckException("Task interrupted while waiting to complete", e);
+        } catch (Exception e) {
+            throw new GoGoDuckException(e.getMessage(), e);
         }
     }
 
-    private static void applySubsetSingleThread(Path tmpDir, GoGoDuckModule module) throws GoGoDuckException {
-        logger.info(String.format("Applying subset on directory '%s'", tmpDir));
-
-        File[] directoryListing = tmpDir.toFile().listFiles();
-        logger.info(String.format("Subset for operation is '%s'", module.getNcksSubsetParameters()));
-
-        for (File file : directoryListing) {
-            applySubsetSingleFileNcks(file, module);
-        }
-    }
-
-    private static void postProcess(Path tmpDir, GoGoDuckModule module) throws GoGoDuckException {
+    private void postProcess(Path tmpDir, GoGoDuckModule module) throws GoGoDuckException {
         File[] directoryListing = tmpDir.toFile().listFiles();
-        for (File file : directoryListing) {
-            module.postProcess(file);
+        if (directoryListing != null) {
+            for (File file : directoryListing) {
+                try {
+                    if (!module.hasPackedVariables()) {
```

Comment:
Why not do this at the start of the method - we know this before doing a directory listing